File: frontend/src/api/practitioner.ts
import { httpClient } from "./http";

export interface ApplyPayload {
  realname: string;
  title: string;
  bio?: string;
  attachment?: string;
}

export interface ApplicationRead {
  id: number;
  realname: string;
  title: string;
  bio?: string | null;
  attachment?: string | null;
  status: string;
  applicant_id: number;
  reviewer_id?: number | null;
  review_note?: string | null;
  created_at: string;
  updated_at: string;
}

export async function applyPractitioner(payload: ApplyPayload): Promise<ApplicationRead> {
  const { data } = await httpClient.post<{ data: ApplicationRead }>("/practitioners/apply", payload);
  return data.data;
}

export async function listApplications(): Promise<ApplicationRead[]> {
  const { data } = await httpClient.get<{ data: ApplicationRead[] }>("/practitioners/applications");
  return data.data;
}

export async function reviewApplication(id: number, decision: "approve" | "reject", review_note?: string): Promise<ApplicationRead> {
  const { data } = await httpClient.post<{ data: ApplicationRead }>(`/practitioners/applications/${id}/review`, { decision, review_note });
  return data.data;
}

File: backend/app/schemas/practitioner.py
"""从业者认证相关 Schema。"""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class PractitionerApply(BaseModel):
    """从业者认证申请请求体。"""

    realname: str = Field(..., min_length=2, max_length=100, description="真实姓名")
    title: str = Field(..., min_length=2, max_length=120, description="职业或头衔")
    bio: Optional[str] = Field(None, max_length=1000, description="个人简介")
    attachment: Optional[str] = Field(None, max_length=500, description="证明材料链接或标识")


class PractitionerReview(BaseModel):
    """管理员审核从业者申请。"""

    decision: str = Field(..., description="approve 或 reject")
    review_note: Optional[str] = Field(None, max_length=500, description="审核意见")


class PractitionerApplicationRead(BaseModel):
    """从业者认证申请响应。"""

    id: int
    realname: str
    title: str
    bio: Optional[str]
    attachment: Optional[str]
    status: str
    applicant_id: int
    reviewer_id: Optional[int]
    review_note: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True
